### PR TITLE
Zycommon integration branch

### LINKED ISSUE #10: Integrating zycommon
Integrate zycommon into the application.

### PR DESCRIPTION
**Pull request by adrianlizarraga**
I integrated the models and services from zycommon and added more ember-paper components. Fixes #10.

signin route:
screen shot 2016-06-10 at 4 09 00 pm

zybooks route:
screen shot 2016-06-10 at 4 07 14 pm

zybook/code route:
screen shot 2016-06-10 at 4 07 28 pm

### CODE REVIEW

**Comment by ssirowy** [comment]
@adrianlizarraga can you merge master into this please?

**Comment by adrianlizarraga** [comment]
@ssirowy merged

File: app/components/zybook-display.js [hunk 1]
```diff
@@ -1,3 +1,7 @@
 import Ember from 'ember';
 
-export default Ember.Component.extend({});
+export default Ember.Component.extend({
+    validAcademicInfo: Ember.computed('zybook.academic_term.{name,year}', function() {
```
Comment (ssirowy) [inline app/components/zybook-display.js]: What does this line do?  Specifically the `{name, year}`

File: app/routes/zybook.js [hunk 2]
```diff
@@ -4,7 +4,13 @@ import AuthenticatedRoute from '../mixins/authenticated-route';
 export default Ember.Route.extend(AuthenticatedRoute, {
     session: Ember.inject.service('session'),
 
+    zyserver: Ember.inject.service('zyserver'),
+
     model: function(params) {
-        return this.get('session.user').getZyBook(params.zybook_code);
+        const bookToFind = JSON.stringify([ params.zybook_code ]);
```
Comment (ssirowy) [inline app/routes/zybook.js]: 👍

File: app/services/session.js [hunk 3]
```diff
@@ -1,64 +1,44 @@
...
     user: null,
 
-    userCredentials: null,
+    credential: null,
 
     isAuthenticated: false,
 
     routing: Ember.inject.service('-routing'),
 
+    zyserver: Ember.inject.service('zyserver'),
```
Comment (ssirowy) [inline app/services/session.js]: When the name of the service is the same name as the property youre defining against, you can use shorthand:
``` javscript
zyserver: Ember.inject.service(),
```

File: app/styles/zybooks.scss [hunk 4]
```diff
@@ -21,14 +21,17 @@
     width: 300px;
     word-wrap: break-word;
 
-    & h3 {
-        margin-left: auto;
-        margin-right: auto;
+    .card-info {
+        float: left;
```
Comment (ssirowy) [inline app/styles/zybooks.scss]: Can you get away with not using `float`?

File: app/templates/zybook.hbs [hunk 5]
```diff
@@ -1,6 +1,11 @@
...
-<div id='toc-container'>
-    {{#each model.chapters as |chapter|}}
-        {{chapter-listing chapter=chapter}}
-    {{/each}}
-</div>
\ No newline at end of file
+<h1 id='zybook-title'>{{model.title}}</h1>
+{{#paper-content class='md-whiteframe-z1'}}
+    {{#paper-list}}
+        {{#each model.chapters as |chapter|}}
```
Comment (ssirowy) [inline app/templates/zybook.hbs]: :+1

**Comment by ssirowy** [comment]
Just a couple of comments. Looking good!

File: app/components/zybook-display.js (hunk 1 above)
Comment (adrianlizarraga) [inline app/components/zybook-display.js]: It's a shorthand way of specifying that the computed property depends on both zybook.academic_term.name and zybook.academic_term.year. 

The dependence on .year is unnecessary in this simple case and should probably be removed.

Comment (ssirowy) [inline app/components/zybook-display.js]: @baileymiller182 fyi.  I didnt know this shorthand existed.   Nice find @adrianlizarraga

File: app/styles/zybooks.scss (hunk 4 above)
Comment (adrianlizarraga) [inline app/styles/zybooks.scss]: Yeah I removed float usage and just decided to use flexbox functionality. I've been using flex for many other components so I should probably be consistent.

File: app/components/zybook-display.js (hunk 1 above)
Comment (adrianlizarraga) [inline app/components/zybook-display.js]: 👍
I didn't have to dig too hard. I found it in the updated tutorial. :)
https://guides.emberjs.com/v2.6.0/object-model/computed-properties/

**Comment by adrianlizarraga** [comment]
I pushed another revision.

**Comment by ssirowy** [comment]
Great job @adrianlizarraga . Nice work.

**Pull request merged**